src/components/transport_manager/src/transport_manager_impl.cc

```diff
@@ -1066,7 +1066,7 @@ void TransportManagerImpl::Handle(TransportAdapterEvent event) {
       LOG4CXX_ERROR(logger_, "Transport adapter failed to send data");
       // TODO(YK): potential error case -> thread unsafe
       // update of message content
-      if (event.event_data.valid()) {
+      if (event.event_data && event.event_data.use_count() > 0) {
```

Comment:
Any reason that `ValidSPtr` isn't used here, but is used in the USB adapter?

Reply:
my bad, that should be ValidSPtr